tests/commonlib/kubernetes.py

```diff
@@ -179,7 +187,83 @@ def delete_resources(self, resource_type: str, **kwargs):
     def patch_resources(self, resource_type: str, **kwargs):
         """
         """
-        return self.dispatch_patch[resource_type](**kwargs)
+        if resource_type == RESOURCE_POD:
+            patch_body = kwargs.pop('body')
+
+            pod = self.get_resource(resource_type, **kwargs)
+            self.delete_resources(resource_type=resource_type, **kwargs)
+            deleted = self.wait_for_resource(resource_type=resource_type, status_list=['DELETED'], **kwargs)
+
+            if not deleted:
+                raise ValueError(f'could not delete Pod: {kwargs}')
+
+            pod = self.create_patched_resource(resource_type, patch_body)
+
+            return pod
+
+        else:
+            return self.dispatch_patch[resource_type](**kwargs)
+    
```

Comment:
we need to simplify logic because python depends on indentations and _if inside if_ is less readable.
in this case we can use 'early termination' technique:

`if resource__type != RESOURCE_POD:
    return self.dispatch_patch[resource_type](**kwargs)`

`patch_body = kwargs.pop('body')`
`...`